### ISSUE TITLE
release/14.x LLD fails to build from source

### ISSUE THREAD

**Issue by andrewrk** [llvm/llvm-project#53553]
Steps to reproduce:

```
cd ~/Downloads
git clone https://github.com/llvm/llvm-project llvm-project-14
cd llvm-project-14
git checkout release/14.x

# LLVM
cd llvm
mkdir build-release
cd build-release
cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/local/llvm14-release -DCMAKE_PREFIX_PATH=$HOME/local/llvm14-release -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_LIBXML2=OFF -G Ninja -DLLVM_PARALLEL_LINK_JOBS=1
ninja install
cd ../..

# LLD
cd lld
mkdir build-release
cd build-release
cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/local/llvm14-release -DCMAKE_PREFIX_PATH=$HOME/local/llvm14-release -DCMAKE_BUILD_TYPE=Release  -G Ninja -DLLVM_PARALLEL_LINK_JOBS=1 -DCMAKE_CXX_STANDARD=17
ninja install
```

Output:

```
[2/58] Building CXX object ELF/CMakeFiles/lldELF.dir/OutputSections.cpp.o
FAILED: ELF/CMakeFiles/lldELF.dir/OutputSections.cpp.o 
/nix/store/apfyw9rd0280rpj2x685s6anyjjnz04l-gcc-wrapper-10.3.0/bin/g++ -DGTEST_HAS_RTTI=0 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/andy/Downloads/llvm-project-14/lld/build-release/ELF -I/home/andy/Downloads/llvm-project-14/lld/ELF -I/home/andy/Downloads/llvm-project-14/lld/include -I/home/andy/Downloads/llvm-project-14/lld/build-release/include -I/include -I/home/andy/local/llvm14-release/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG  -fno-exceptions -fno-rtti -MD -MT ELF/CMakeFiles/lldELF.dir/OutputSections.cpp.o -MF ELF/CMakeFiles/lldELF.dir/OutputSections.cpp.o.d -o ELF/CMakeFiles/lldELF.dir/OutputSections.cpp.o -c /home/andy/Downloads/llvm-project-14/lld/ELF/OutputSections.cpp
/home/andy/Downloads/llvm-project-14/lld/ELF/OutputSections.cpp:19:10: fatal error: llvm/Config/config.h: No such file or directory
   19 | #include "llvm/Config/config.h" // LLVM_ENABLE_ZLIB
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[3/58] Building CXX object MachO/CMakeFiles/lldMachO.dir/Driver.cpp.o
FAILED: MachO/CMakeFiles/lldMachO.dir/Driver.cpp.o 
/nix/store/apfyw9rd0280rpj2x685s6anyjjnz04l-gcc-wrapper-10.3.0/bin/g++ -DGTEST_HAS_RTTI=0 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/andy/Downloads/llvm-project-14/lld/build-release/MachO -I/home/andy/Downloads/llvm-project-14/lld/MachO -I/home/andy/Downloads/llvm-project-14/lld/include -I/home/andy/Downloads/llvm-project-14/lld/build-release/include -I/include -I/home/andy/local/llvm14-release/include -I/../libunwind/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG  -fno-exceptions -fno-rtti -MD -MT MachO/CMakeFiles/lldMachO.dir/Driver.cpp.o -MF MachO/CMakeFiles/lldMachO.dir/Driver.cpp.o.d -o MachO/CMakeFiles/lldMachO.dir/Driver.cpp.o -c /home/andy/Downloads/llvm-project-14/lld/MachO/Driver.cpp
In file included from /home/andy/Downloads/llvm-project-14/lld/MachO/Driver.cpp:23:
/home/andy/Downloads/llvm-project-14/lld/MachO/UnwindInfoSection.h:16:10: fatal error: mach-o/compact_unwind_encoding.h: No such file or directory
   16 | #include "mach-o/compact_unwind_encoding.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[19/58] Building CXX object ELF/CMakeFiles/lldELF.dir/SyntheticSections.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by MaskRay** [comment]
Duplicate of #48572 

But please paste the message there since the log is useful.

**Comment by andrewrk** [comment]
Can you put the other one in the release/14.x milestone?

**Comment by tstellar** [comment]
I think this is a different failure than https://github.com/llvm/llvm-project/issues/48572.

**Comment by tstellar** [comment]
Well at least the llvm-config.h failure is new.

**Comment by nikic** [comment]
Might be related to https://reviews.llvm.org/D116492, in which case https://reviews.llvm.org/D118792 might help.

**Comment by MaskRay** [comment]
I created https://reviews.llvm.org/D119380 to deprecate the standalone build following libc++/libc++abi/libunwind.

**Comment by MaskRay** [comment]
llvm/Config/config.h has been fixed. Closing.

**Comment by tstellar** [comment]
It looks like the backport that fixed this bug was 923b67e36917352c1ca1fb1e339dbe5acdc97ddf.
